File: Objective-C/WholeSquare.m
//
//  WholeSquare.m
//  Codility-Tests
//
//  Created by Mohamed EL Meseery on 11/29/16.
//
//

#import <Foundation/Foundation.h>

/**
    Refer to WholeSquare challenge file for description.
 **/

int solution(int A, int B)  {
    
    int count = 0;
    NSMutableArray * range = [NSMutableArray array];
    
    int lower = A<B ? A : B;
    int upper = A>B ? A : B;
    
    for (int i = lower; i<=upper; i++) {
        [range addObject:@(i)];
    }
    
    for (NSNumber *i in range){
        double square = sqrt(i.doubleValue);
        int sq = (int)square;
        if (square == (double)sq) {
            count++;
        }
    }
    
    return count;
}

File: Swift/WholeSquare.swift
//
//  WholeSquare.swift
//  Codility-Tests
//
//  Created by Mohamed EL Meseery on 11/29/16.
//
//

import Foundation

/**
    Refer to WholeSquare challenge file for description.
 **/

public func solution(_ A : Int, _ B : Int) -> Int {
    var count = 0
    var range = 0...0
    
    if A>B {
        range = B...A
    }else{
        range = A...B
    }
    
    for i in range {
        let square = sqrt(Double(i))
        if  square == Double(Int(square)){
            count = count + 1
        }
    }
    return count
}
    
